packages/salesforce-adapter/src/adapter.ts

```diff
@@ -349,6 +351,9 @@ export default class SalesforceAdapter implements AdapterOperations {
 
     const fetchProfile = buildFetchProfile(config.fetch ?? {})
     this.fetchProfile = fetchProfile
+    if (this.fetchProfile.isFeatureEnabled('fetchCustomObjectUsingRetrieveApi')) {
```

Comment:
```suggestion
    if (!this.fetchProfile.isFeatureEnabled('fetchCustomObjectUsingRetrieveApi')) {
```

If the feature is disabled we want to take it out, not?